ambry-rest/src/main/java/com.github.ambry.rest/NettyResponseChannel.java

```diff
@@ -148,6 +148,18 @@ public void onCompletion(Long result, Exception exception) {
     if (!responseMetadataWriteInitiated.get()) {
       maybeWriteResponseMetadata(responseMetadata, new ResponseMetadataWriteListener());
     }
+    if (finalResponseMetadata == null) {
+      // If finalResponseMetadata is still null, it indicates channel becomes inactive.
+      if (ctx.channel().isActive()) {
+        logger.warn("Channel should be inactive status. {}", ctx.channel());
```

Comment:
Could you add a metric for this? We can set up an alert if this case ever occurs.